### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 1]
```diff
@@ -8731,6 +8731,7 @@ static const CK_MECHANISM_TYPE ep11_supported_mech_list[] = {
     CKM_SHA_1_HMAC,
     CKM_GENERIC_SECRET_KEY_GEN,
     CKM_IBM_ATTRIBUTEBOUND_WRAP,
+    CKM_GENERIC_SECRET_KEY_GEN,
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_specific.c:8762]: It was there, two lines above

Comment (ifranzki) [reply usr/lib/ep11_stdll/ep11_specific.c:8762]: Arg, you are right. It was already added by https://github.com/opencryptoki/opencryptoki/commit/945a46aecd3006387e551a76d8d94ea622bddc4a
I guess this was a git-rebase error to add it again. 
SO I'll drop this commit entirely.

File: usr/lib/common/asn1.c [hunk 2]
```diff
@@ -4215,12 +4215,28 @@ CK_RV ber_decode_IBM_DilithiumPrivateKey(CK_BYTE *data,
...
+            TRACE_DEVEL("ber_decode_CHOICE returned invalid option %ld\n",
+                        option);
+            goto cleanup;
+        }
+
+        offset += field_len - len;
+
+        rc = ber_decode_BIT_STRING(buf + offset, &tmp, &len, &field_len);
+        if (rc != CKR_OK) {
+            TRACE_DEVEL("ber_decode_BIT_STRING of (t1) failed\n");
```
Comment (jschmidb) [inline usr/lib/common/asn1.c:4254]: What happens if option == 0, but there is no (t1) value given? Will field_len be just 0 and the build_attribute below creates a "dummy" attr with a length of 0? Should we probably don't add the attr at all if (t1) is not given?

Comment (ifranzki) [reply usr/lib/common/asn1.c:4254]: Well, the same question applies to all the key components, even those that are not optional by design. If their length would be zero we would add a zero length attribute (build_attribute can deal with len = 0).

That would anyway be an invalid encoding of an Dilithium key. 
So: garbage in, garbage out :-)

**Comment by jschmidb** [comment]
OK .... other commits lgtm.

**Review by jschmidb — APPROVED** [review]

**Pull request merged**